### DISCUSSION TITLE
Email deliverability to microsoft for auth/v1/verify endpoint

### DISCUSSION THREAD

**Discussion by silverbackdan** [supabase/supabase#19483] [Questions]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I have been doing extensive testing from the Amazon SES dashboard. Changing the from email address, checking authentication and email contents to figure out why there are deliverability issues and have found that sometime over the weekend it appears URLs using `supabase.com/auth/v1/verify` (including any subdomain) are being filtered or not delivered to mailboxes on Microsoft systems. Emails to O365 mailboxes are undelivered and @outlook.com mailboxes are going to junk.

This means deliverability of users trying to sign in with OTP is severely affected.

## To Reproduce

Examples of sending an email to a microsoft mailbox with this content fails or goes to junk
```
Hello my email message from the test system

https://supabase.co/auth/v1/verify
```

```
Hello my email message from the test system

https://anything.supabase.co/auth/v1/verify
```

```
Hello my email message from the test system

https://anything.supabase.co/auth/v1/verify?auth=any_query
```

These examples are delivered without issues
```
Hello my email message from the test system

https://supabase.co/auth/v1/verif
```

```
Hello my email message from the test system

https://anything.co/auth/v1/verify
```

```
Hello my email message from the test system

https://anything.co/verify
```

```
Hello my email message from the test system

https://anything.co/auth/verify
```

## Expected behavior

This is an infrastructure issue. Emails should be delivered. I have raised a critical issue via the support methods.

## Screenshots

n/a

## System information

- OS: [e.g. macOS, Windows]
- Browser (if applies) [e.g. chrome, safari]
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by hf** [comment]
Unfortunately, this is not a Supabase issue, although it certainly sounds like one. Email deliverability is a very complex topic with many moving parts and virtually zero visibility on why certain "junk" decisions are being made by email systems.

There's really not much we can do systemically to solve this issue. You can however improve your reliability by:

1. Enabling DKIM, DMARC and SPF on the domain name you're using in the "From" field. This has to be done in concert with your email sending provider. [Docs for AWS SES.](https://docs.aws.amazon.com/ses/latest/dg/email-authentication-methods.html)
2. Customize the email templates to be different than the included templates from Supabase.
3. Use [custom domains](https://supabase.com/docs/guides/platform/custom-domains) or [vanity subdomains](https://supabase.com/docs/guides/platform/custom-domains#vanity-subdomains) for your Supabase project to avoid relying on `project-ref.supabase.co` URLs.

I wish there was something we could do about this, but unfortunately everything is out of our control.

**Comment by silverbackdan** [comment]
Hello,

Thank you for your reply, however I’d beg to differ and say it certainly is a Supabase issue.

I believe it will cause a huge reliability issue for people using any email login methods using magic links as it is purely the link back to Supabase verify endpoint which triggers this spam result.

I indeed have DMARC, DKIM and SPF and have validated all of my settings. Everything had been working just fine and I promise you, with all of my testing, it is simply the supabase domain name with the verify auth endpoint which causes the issue.

I have done so much A/B testing and nothing I do to the emails except for change the domain or verify endpoint path (and I don’t just mean change the subdomain because I’d simply setup a new project) – nothing will stop this new issue which has just been triggered.

This means, anyone who does not wish to opt for a pain plan with the custom domains add-on will not reliably be able to allow users on Microsoft’s network to login using magic links. Any recipient domain using the O365 mailboxes, which are very popular, will not receive the emails at all.

I believe it’s an issue your team should likely raise with Microsoft directly, so I’d appreciate if you could escalate this issue to the team who will deal with this type of problem.

I appreciate it isn’t an easy problem to resolve – nor one I envy your team having to look into. I still think it’s a serious issue nonetheless and one which should be investigated with urgency as to why this link is now triggering emails to either not be delivered, or go into the junk folder.

Thanks again, I really do appreciate your response – but I feel it’s important to be really direct here just due to the severity of the issue, and I’ve done a lot of work in the past with email deliverability so, I know, it’s not a nice thing to have to look into. Hopefully though, should the correct appeals be raised with Microsoft, it can be something you develop a closer relationship with them on should this happen again.

Many thanks,

Daniel

**Comment by hf** [comment]
I can try to talk to Microsoft, but this is probably best approached from your side if you have a relationship with them via Office 365.

We have no relationship to Microsoft at this time, nor to any server that receives email in the world. So although we can try (somehow) with larger providers like Microsoft, I can't promise any sort of outcome as it's out of our hands.

I can only say that by using the custom domain solution, the problem is likely to completely go away. You can also set up your own proxy if the pricing of that Supabase feature does not meet your expectations or budget.

**Comment by silverbackdan** [comment]
Thanks for taking the time to understand and reply. I can contact them regarding with accounts I have where I cannot get the email to even reach the junk folder and see what happens there. We will purchase the paid plan for now with the custom domain for this project. In future, what would the proxy option be? Do you have an example?

I hope there must be some form of appeal process for this kind of link blocking/filtering in the Microsoft ecosystem.

It's so strange it is only when the supabase.co domain and the path to the verify endpoint is used. I imagine there must have been some malicious activity picked up to cause this.

Thanks again.

**Comment by hf** [comment]
Sorry, one more thing that I don't think is quite obvious from the discussion.

You actually _don't_ have to use the Supabase Auth URL _at all_ in the email template. You can always set up your own web page, let's say `https://example.com/magiclink` where you can certainly use the Supabase Auth APIs to sign in a user.

For example, by setting a template such as:

```html
<h2>Magic Link</h2>

<p>Follow this link to login:</p>
<p><a href="https://example.com/magiclink?token_hash={{ .TokenHash }}">Log In</a></p>
```

And by calling [`supabase.auth.verifyOtp`](https://supabase.com/docs/reference/javascript/auth-verifyotp) with the hash value from the URL in the browser, you will achieve the same effect as if you had a custom domain without having to pay us at all.

I am noting these things however so that we can address them in docs and in the platform itself, so thank you for the feedback! It is really invaluable!

**Comment by silverbackdan** [comment]
This is a fantastic workaround - I was not aware of this option. We'll get cracking with this solution as a workaround for now.

**Comment by hf** [comment]
OK if I convert this to a discussion, so it's easier to find if someone has a similar problem in the future?

**Comment by silverbackdan** [comment]
Sure, that's fine by me. It may crop up over the next few weeks if Microsoft don't change their filtering. It only started happening over this weekend so, yes, useful if anyone needs to get a solution in place.

**Comment by silentworks** [comment]
The option @hf mentioned is somewhat covered here https://supabase.com/docs/guides/auth/auth-email-templates#redirecting-the-user-to-a-server-side-endpoint but we also have another guide about this option but we just haven't merged it in as yet https://github.com/supabase/supabase/pull/16683

**Comment by kangmingtay** [comment]
@silverbackdan just to add on, if you're using custom domains, you can [initialise the supabase client to use the custom domain](https://supabase.com/docs/guides/platform/custom-domains#activate-your-domain) - which means that it will be used in the `/verify` endpoint sent in the email instead of the default one that supabase provides (which seems to be denied by O365)

**Comment by silverbackdan** [comment]
Ah ha, thanks - this is very useful - also in the solution to bypass the link to supabase verify endpoint and verify it in my app. Because in development it's useful for the domain to be localhost. Thanks.
